test/features/deploy_kogito_runtime.feature

```diff
@@ -9,12 +9,11 @@ Feature: Deploy Kogito Runtime
   Scenario Outline: Deploy Simplest Scenario using Kogito Runtime
     Given Kogito Operator is deployed
     And Clone Kogito examples into local directory
-    And Local example service "<example-service>" is built by Maven using profile "<profile>"
-    And Local example service "<example-service>" is deployed to image registry, image tag stored as variable "built-image"
+    And Local example service "<example-service>" is built by Maven using profile "<profile>" and deployed to registry if not available, image tag stored as variable "built-image"
 
-    When Deploy <runtime> example service using image in variable "built-image" with configuration:
+    When Deploy <runtime> example service "<example-service>" using image in variable "built-image" with configuration:
```

Comment:
I would rewrite these steps to: 
- "Local example service "<example-service>" is built by Maven using profile "<profile>" and deployed to registry"
Where internally, the variable is called "built-image-<example-service>"
- "Deploy <runtime> example service "<example-service>" from registry with configuration:"
Where from registry will search the above variable transparently

Reply:
Problem is that to resolve image name you have to know maven profiles used to build the image.
And I would like to avoid passing maven profiles to deploy step as it is irrelevant there

Reply:
I meant to keep using the variable with the resolved image name, but without needing to pass the variable name, but having like a contract between these two steps.

Reply:
So you mean to store built image to property named like "<service name>-runtime-image" where service name will be a key for resolving image name?

From my perspective:
- pros - Simpler step syntax
- cons - Hiding of dependency between steps (casual reader won't realize that these steps are connected and how)
           - "Artificial dependency" - deployed service has to have exactly same name as built service folder name
           - A bit more logic in the step implementation (more code to take care of :) )

I would incline to current solution, though I would be ok with either of those.
@radtriste What do you think?

Reply:
Thinking that we should always be explicit for readability to any user.
The writer of the feature may be not correlated to the writer of the implementation and the implementation could be changed later, forgetting this dependency.
Having it explicitly defined in the feature helps to understand what is happening really in the scenario, without having to dig in the code.
I think we should be explicit whenever we can.

Reply:
I disagree that we're hiding dependencies between steps:
- As far I can see it, saying "to registry" and "from registry" is enough and understandable for readers. What I mean is that saying "to registry with variable "X"" and "from registry using variable "X"", the variable "X" is a bit verbose for me as this should be something internal for the registry part and the service to be deployed in the current scenario.

- Also, about the logic in the step implementation, take into account that the variables are stored in the scenario context, so we don't need to deal with parallel execution. Therefore, it's a matter to use the variable name "builtImage" or to use the variable name "<service name>". Don't see this overcomplicate the logic here.

Reply:
That can be fine for one image being used in one scenario.
But what about having for example 3 deployments in one scenario (viz service discovery)?:

Local example service "onboarding-example/hr" ... and deployed to registry"
Local example service "onboarding-example/payroll" ... and deployed to registry"
Local example service "onboarding-example/onboarding-quarkus" ... and deployed to registry"

Deploy example service "hr" from registry...
Deploy example service "payroll" from registry...
Deploy example service "onboarding-quarkus" from registry...

Reply:
There would be three variables, one for each service. data.ScenarioContext[imagePrefix + serviceName].

Reply:
yes, that means that reader needs to realize "built folder name" == "deployed service name"

Reply:
Why? the reader only knows that the service is in the registry, and that it will used afterwards.

Reply:
And how does reader know that for example "hr" service will be in the registry and for example "management" service doesn't exist there?

Reply:
Do you concern is that the word "registry" is too open? Maybe it is. What about to say "runtime registry" instead of just "registry"? 

```
Local example service "onboarding-example/hr" ... and deployed to runtime registry"
Deploy example service "hr" from runtime registry...
```

This way it does also reference to the runtime registry arguments as well.

Reply:
We are already using some internal variables, for example for the local checkout of project and build, and so some steps are already dependent on each other.
I would go for the internal var instead of "in step" description. We habe been using the `service name` as a key for most of our steps until now and we could continue with it.
Moreover, we have a "default" behavior if the step `Local example service ...` is not existing: we would be generating the image's name with our code anyway, so it would be easy to retrieve from `runtime registry` (I would add the `runtime` like suggested by @Sgitario to be more explicit where we take the image from and that this is happening).

Finally, I think we should start to use internal var if we have different images for the same service. If we have only one image per service, it is easy. But I don't think we need this, at least for now ...